src/backend/api/meals.js

```diff
@@ -5,11 +5,70 @@ const knex = require("../database");
 router.get("/", async (request, response) => {
   try {
     // knex syntax for selecting things. Look up the documentation for knex for further info
-    const titles = await knex("meals").select("title");
-    response.json(titles);
+    const maxPrice = request.query.maxPrice;
+    const title = request.query.title;
+    const createdAfter = request.query.createdAfter;
+    const limit = request.query.limit;
+    const availableReservations = request.query.availableReservations;
+    let allMeals = await knex("meal")
+      .leftJoin("reservation", "meal.id", "=", "reservation.meal_id")
+      .select("*");
+    if (maxPrice || title || createdAfter || limit || availableReservations) {
+      if (maxPrice) {
+        allMeals = allMeals.filter((meal) => meal.price < parseInt(maxPrice));
+      }
+      if (title) {
+        allMeals = allMeals.filter((meal) =>
+          meal.title.toLowerCase().includes(title.toLowerCase())
+        );
+      }
+      if (createdAfter) {
+        allMeals = allMeals.filter(
+          (meal) => Date.parse(meal.created_date) >= Date.parse(createdAfter)
+        );
+      }
+      if (limit) {
+        allMeals = allMeals.slice(0, limit);
+      }
+      if (availableReservations) {
+        allMeals = allMeals.filter(
+          (meal) => meal.max_reservations > meal.number_of_guests
```

Comment:
you won't have number_of_guests in the meal table. You need to join the reservations table, sum the total reservations and then compare them with the max_reservations in order to achieve the above query.